sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/StorageHelperTests.cs

```diff
@@ -11,17 +12,53 @@ namespace Azure.Monitor.OpenTelemetry.Exporter.Tests
 {
     public class StorageHelperTests
     {
-        // The directory separator is either '/' or '\' depending on the Platform, making unit testing tricky.
-        private static readonly char ds = Path.DirectorySeparatorChar;
+        private readonly string _testStorageDirectory;
+
+        public StorageHelperTests()
+        {
+            _testStorageDirectory = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? @"C:\UnitTest"
+                : "/var/UnitTest";
+        }
 
         [Fact]
-        public void VerifyConfiguredDirectory()
+        public void VerifyGetStorageDirectory_Configured()
+        {
+            var directoryPath = StorageHelper.GetStorageDirectory(
+                platform: new MockPlatform
+                {
+                    UserName = "ApplicationPoolIdentity",
+                    ProcessName ="w3wp",
+                    ApplicationBaseDirectory = "C:\\inetpub\\wwwroot",
+                },
+                configuredStorageDirectory: _testStorageDirectory,
+                instrumentationKey: "testIkey");
+
+            var expected = Path.Combine(_testStorageDirectory, "f01a11b594e1f6d06cd96564b1f258b515bf98d956e8a1842c74479a1ecef4eb");
+            Assert.Equal(expected, directoryPath);
+        }
+
+        [Theory]
+        [InlineData("WINDOWS", "LOCALAPPDATA")]
+        [InlineData("LINUX", "TMPDIR")]
+        public void VerifyGetStorageDirectory_Default(string osName, string envVarName)
         {
+            var platform = new MockPlatform
+            {
+                UserName = "ApplicationPoolIdentity",
+                ProcessName = "w3wp",
+                ApplicationBaseDirectory = "C:\\inetpub\\wwwroot",
```

Comment:
Little difficult to follow here. I understand its a mock, but seeing a c:\ for Linux tests confuses.

Reply:
This started as the Windows example, and all three values would be familiar to someone with IIS experience.
It's not immediately clear from reading the test, but these values are only used to calculate the hash used in the final storage directory.

Any changes here for a Linux based example would also require a new hash value. My gut feeling was that would over complicate the test.

But I can make that change to make the test easier to read and understand.